components/edit-collective/sections/EditCollectivePage.js

```diff
@@ -212,9 +243,11 @@ const EditCollectivePage = ({ collective }) => {
                         setSections(getNewSections(sections, dragIndex, hoverIndex));
                         setDirty(true);
                       }}
-                      onSectionToggle={(selectedSection, isEnabled) => {
+                      onSectionToggle={(selectedSection, isEnabled, restrictedTo) => {
                         const sectionIdx = sections.findIndex(({ section }) => section === selectedSection);
-                        const newSections = set(cloneDeep(sections), `${sectionIdx}.isEnabled`, isEnabled);
+                        let newSections = cloneDeep(sections);
+                        newSections = set(newSections, `${sectionIdx}.isEnabled`, isEnabled);
+                        newSections = set(newSections, `${sectionIdx}.restrictedTo`, restrictedTo);
```

Comment:
`set` mutates the object, so we can simplify

```suggestion
                        const newSections = cloneDeep(sections);
                        set(newSections, `${sectionIdx}.isEnabled`, isEnabled);
                        set(newSections, `${sectionIdx}.restrictedTo`, restrictedTo);
```